spec/lib/msf/ui/console/command_dispatcher/core_spec.rb

```diff
@@ -397,4 +397,96 @@ def set_tabs_test(option)
+          'Active sessions',
+          '===============',
+          '',
+          'No active sessions.'
+        ])
+      end
+    end
+
+    context "filtering with search results" do
+      before do
+        allow(framework).to receive(:sessions).and_return(sessions)
+        allow(double "Session").to receive(:kill)
+      end
+
+      let(:sessions) do 
+        {
+          1 => double('Session', last_checkin: Time.now, type: 'meterpreter', sid:1, sname: "sesh1", info: "info", session_host: "127.0.0.1", tunnel_to_s: "tunnel"),
+          2 => double('Session', last_checkin: (Time.now - 90), type: 'meterpreter', sid:2, sname: "sesh2", info: "info", session_host: "127.0.0.1", tunnel_to_s: "tunnel"),
+          3 => double('Session', last_checkin: Time.now, type: 'java', sid:3, sname: "sesh3", info: "info", session_host: "127.0.0.1", tunnel_to_s: "tunnel")
+        }
+      end
+
+      it "filters by session_id" do
+        core.cmd_sessions("--search", "session_id:2")
+        expect(@output.join("\n")).to match_table <<~TABLE
+        Active sessions
+        ===============
+ 
+          Id  Name   Type         Information  Connection
+          --  ----   ----         -----------  ----------
+          2   sesh2  meterpreter  info         tunnel (127.0.0.1)
+        TABLE
+      end
+
+      it "filters by last checkin" do
+        core.cmd_sessions("--search", "last_checkin:after:10s")
+        expect(@output.join("\n")).to match_table <<~TABLE
+        Active sessions
+        ===============
+ 
+          Id  Name   Type         Information  Connection
+          --  ----   ----         -----------  ----------
+          2   sesh2  meterpreter  info         tunnel (127.0.0.1)
+        TABLE
+      end
+
+      it "filters by session type" do
+        core.cmd_sessions("--search", "session_type:java")
+        expect(@output.join("\n")).to match_table <<~TABLE
+        Active sessions
+        ===============
+ 
+          Id  Name   Type  Information  Connection
+          --  ----   ----  -----------  ----------
+          3   sesh3  java  info         tunnel (127.0.0.1)
+        TABLE
+      end
+
+      it "Killall should not kill anything if nothing matches" do
+        core.cmd_sessions("--search", "session_id:5", "-K")
+        expect(@output).to eq([
+          'No matching sessions.'
+        ])
+      end
+
+      it "Killall should kill only the specified sessionS" do
+        expect(sessions[2]).to receive(:kill)
+        core.cmd_sessions("--search", "session_id:2 session_id:3", "-K")
+        expect(@output.join("\n")).to match_table <<~TABLE
+        Killing matching sessions...
+        Active sessions
+        ===============
+
+          Id  Name   Type         Information  Connection
+          --  ----   ----         -----------  ----------
+          2   sesh2  meterpreter  info         tunnel (127.0.0.1)
+          3   sesh3  java         info         tunnel (127.0.0.1)
+        TABLE
```

Comment:
heredoc can be aligned/extra indented: https://www.rubyguides.com/2018/11/ruby-heredoc/ for readability, similar to 

https://github.com/rapid7/metasploit-framework/blob/1ee7f03a2b5031f88de37af7b135ec4d83a827f2/spec/lib/msf/base/serializer/readable_text_spec.rb#L111-L117